### PR TITLE
MM-49235 - Fix eslint warnings, better TS typing

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Mostly annoying little things, but some good TS typings to help us in the future.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-49235

### CODE REVIEW

File: standalone/src/init.ts [hunk 1]
```diff
@@ -254,49 +265,51 @@ export default async function init(cfg: InitConfig) {
     connectCall(channelID, callTitle, getWSConnectionURL(getConfig(store.getState())), iceConfigs, (ev) => {
         switch (ev.event) {
         case 'hello':
-            setServerVersion(ev.data.server_version)(store.dispatch, store.getState);
+            // TODO: where is this server_version coming from? I think we can remove this case.
+            // @ts-ignore
+            store.dispatch(setServerVersion(ev.data.server_version));
```
Comment (cpoile) [inline standalone/src/init.ts:270]: This is an open question. Would this event even reach here? I don't think so...?

Comment (streamer45) [reply standalone/src/init.ts:270]: The initial `hello` websocket response [contains](https://github.com/mattermost/mattermost-server/blob/3c1f63d41b7ac57699ea7945db17f22fb19f7c8d/app/platform/web_conn.go#L664) the server version which as far as I remember was necessary to be in store to make things function properly. Mostly mimicking what happens in [webapp](https://github.com/mattermost/mattermost-webapp/blob/e66cfe689f501a919af5be1dc16f0a01d3da6fd7/actions/websocket_actions.jsx#L1244-L1247).

Comment (streamer45) [reply standalone/src/init.ts:270]: As to why we get there, that would be through https://github.com/mattermost/mattermost-plugin-calls/blob/e2d24282ec8a3ed0292abc2328e6d742bdf00232/webapp/src/websocket.ts#L116

Comment (cpoile) [reply standalone/src/init.ts:270]: That's funny, because I logged it and I never reached it... Shouldn't it fire on every new call?

Comment (streamer45) [reply standalone/src/init.ts:270]: I'd expect it to since we are creating a brand new connection, yes.

Comment (cpoile) [reply standalone/src/init.ts:270]: Just double checked -- it doesn't seem to be receiving it.

Comment (streamer45) [reply standalone/src/init.ts:270]: Honestly I cannot reproduce this, the event handler is always registered before connection happens and the `hello` event emitted correctly. 

Could you verify whether in your case the event handler is registered before or not?

File: standalone/src/init.ts [hunk 2]
```diff
@@ -174,11 +189,11 @@ async function fetchChannelData(store: Store, channelID: string) {
                 },
             });
 
-            const userStates = {} as any;
+            const userStates: Record<string, UserState> = {};
             const users = resp.call.users || [];
             const states = resp.call.states || [];
             for (let i = 0; i < users.length; i++) {
-                userStates[users[i]] = states[i];
+                userStates[users[i]] = {...states[i], id: users[i], voice: false};
```
Comment (cpoile) [inline standalone/src/init.ts:196]: Added some fields here. Wanted to point it out.

**Review by streamer45 — APPROVED** [review]
Awesome, let's merge this before it causes more pain :)

**Comment by cpoile** [comment]
@streamer45 Thanks!

**Pull request merged**